### PR TITLE
Add Initial Implementation of the `StateMultipleChoiceQuestionPage` [WIP]

### PR DESCRIPTION
**Pull request by law-levi-san**

### CODE REVIEW

File: components/StateMultipleChoiceQuizPage.tsx [hunk 1]
```diff
@@ -1,10 +1,57 @@
...
+  correctAnswer: string;
+};
+const quizQuestions: Question[] = [
+  {
+    questionText: "What is Karnataka's capital?",
+    option: ["Bangalore", "Chennai", "Hyderabad", "Raipur"],
+    correctAnswer: "Bangalore",
+  },
+];
+
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Move this to a separate file in the `types` folder.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 2]
```diff
@@ -1,10 +1,57 @@
...
+const StateMultipleChoiceQuizApp: React.FC = () => {
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: In other files, the syntax we used is different. Let's be consistent.
```suggestion
export default function StateMutipleChoiceQuizPage() {
```

File: components/StateMultipleChoiceQuizPage.tsx [hunk 3]
```diff
@@ -1,10 +1,57 @@
...
+  const [currentQuestionIndex, setCurrentQuestion] = useState<number>(0);
+  const [score, setScore] = useState<number>(0);
+  const [quizFinished, setQuizFinished] = useState<boolean>(false);
+
+  const currentQuestion = quizQuestions[currentQuestionIndex];
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Club all these `const`s without line breaks.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 4]
```diff
@@ -1,10 +1,57 @@
...
+    if (selectedOption === currentQuestion.correctAnswer) {
+      setScore(score + 1);
+    }
+    if (currentQuestionIndex < quizQuestions.length - 1) {
+      setCurrentQuestion(currentQuestionIndex + 1);
+    } else {
+      setQuizFinished(true);
+    }
+  };
   return (
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Add a line break before the `return` statement.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 5]
```diff
@@ -1,10 +1,57 @@
...
+              onPress={() => handleAnswer(option)}
+            ></Button>
+          ))}
+        </View>
+      )}
     </View>
   );
-}
+};
+export default StateMultipleChoiceQuizApp;
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Add a line break before the `export` statement.

**Review by Kicchu02 — CHANGES_REQUESTED** [review]
@law-levi-san You have added a single question. I would like you to add a few more questions to fully understand how the flow of the quiz works.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 6]
```diff
@@ -1,10 +1,57 @@
...
+        </View>
+      ) : (
+        <View>
+          <Text>{currentQuestion.questionText}</Text>
+          {currentQuestion.option.map((option) => (
+            <Button
+              key={option}
+              title={option}
+              onPress={() => handleAnswer(option)}
+            ></Button>
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]:
```suggestion
             / >
```

File: components/StateMultipleChoiceQuizPage.tsx [hunk 7]
```diff
@@ -1,10 +1,57 @@
-import React from "react";
-import { Text, View } from "react-native";
+import { useTheme } from "@react-navigation/native";
+import React, { useState } from "react";
+import { Text, View, Button } from "react-native";
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Optimize imports.

File: components/StateMultipleChoiceQuizPage.tsx (hunk 1 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have in the recent commit, pls check

File: components/StateMultipleChoiceQuizPage.tsx (hunk 2 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have made this file consistent with other files by adding the export statement while defining the function. Please check

File: components/StateMultipleChoiceQuizPage.tsx (hunk 3 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have, see once, pls let me know if i have missed out on these yet again

File: components/StateMultipleChoiceQuizPage.tsx (hunk 4 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have in the recent commit

File: components/StateMultipleChoiceQuizPage.tsx (hunk 5 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have removed the export statement at the bottom to the top, where the function is being defined, in the recent commit. Check once

File: components/StateMultipleChoiceQuizPage.tsx (hunk 6 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have made this change. But didn't understand why is there a necessity for a self-closing tag, kindly explain.

File: components/StateMultipleChoiceQuizPage.tsx (hunk 7 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have in the recent commit, pls check.

**Comment by law-levi-san** [comment]
> @law-levi-san You have added a single question. I would like you to add a few more questions to fully understand how the flow of the quiz works.

I have added 13 more such questions in the recent commit. Pls check once

File: components/StateMultipleChoiceQuizPage.tsx [hunk 8]
```diff
@@ -1,10 +1,55 @@
...
+        <View>
+          <Text>{currentQuestion.questionText}</Text>
+          {currentQuestion.option.map((option) => (
+            <Button
+              key={option}
+              title={option}
+              onPress={() => handleAnswer(option)}
+            />
+          ))}
+        </View>
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Can you make these two as separate components in the same file? Refer to the example file I shared with you.

File: types/Question.tsx
Comment (Kicchu02) [inline types/Question.tsx:1]: I told to change the file name to `QuizQuestions.tsx` in whatsapp.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 9]
```diff
@@ -1,10 +1,55 @@
...
+  const handleAnswer = (selectedOption: string) => {
+    if (selectedOption === currentQuestion.correctAnswer) {
+      setScore(score + 1);
+    }
+    if (currentQuestionIndex < quizQuestions.length - 1) {
+      setCurrentQuestion(currentQuestionIndex + 1);
+    } else {
+      setQuizFinished(true);
+    }
+  };
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Rearrange these properly.
```suggestion
  const [quizFinished, setQuizFinished] = useState<boolean>(false);
  const [shuffledQuestions, setShuffledQuestions] = useState<Question[]>([]);
  const currentQuestion: Question = shuffledQuestions[currentQuestionIndex] || {
    questionText: "",
    option: [],
    correctAnswer: "",
  };

  useEffect(() => {
    const shuffled = [...quizQuestions].sort(() => Math.random() - 0.5);
    setShuffledQuestions(shuffled);
  }, []);

  const handleAnswer = (selectedOption: string) => {
    if (selectedOption === currentQuestion.correctAnswer) {
      setScore(score + 1);
    }
    if (currentQuestionIndex < quizQuestions.length - 1) {
      setCurrentQuestion(currentQuestionIndex + 1);
    } else {
      setQuizFinished(true);
    }
  };
  
```

File: components/StateMultipleChoiceQuizPage.tsx [hunk 10]
```diff
@@ -1,10 +1,55 @@
...
+export default function StateMultipleChoiceQuizApp() {
+  const [currentQuestionIndex, setCurrentQuestion] = useState<number>(0);
+  const [score, setScore] = useState<number>(0);
+  const [quizFinished, setQuizFinished] = useState<boolean>(false);
+  const [shuffledQuestions, setShuffledQuestions] = useState<Question[]>([]);
+
+  useEffect(() => {
+    const shuffled = [...quizQuestions].sort(() => Math.random() - 0.5);
+    setShuffledQuestions(shuffled);
+  }, []);
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: I don't think so this logic is working. I am getting all 14 questions while running the app.

**Review by Kicchu02 — CHANGES_REQUESTED** [review]
@law-levi-san Please address all the changes requested in the previous review and offline, WhatsApp etc so that I don't have to check the same thing again and again.

File: components/StateMultipleChoiceQuizPage.tsx (hunk 10 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: Its working fine when I run the app, shuffled questions too. Kindly check once again.

File: components/StateMultipleChoiceQuizPage.tsx (hunk 9 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have in the recent commit, pls check. If there's any more rearrangements needed, kindly let me know.

File: types/Question.tsx
Comment (law-levi-san) [reply types/Question.tsx:1]: I have renamed the file in the latest commit, pls check once

File: components/StateMultipleChoiceQuizPage.tsx (hunk 8 above)
Comment (law-levi-san) [reply components/StateMultipleChoiceQuizPage.tsx]: I have, as directed in the file shared with me, in the recent commit.

File: components/StateMultipleChoiceQuizPage.tsx [hunk 11]
```diff
@@ -1,10 +1,57 @@
-import React from "react";
-import { Text, View } from "react-native";
+import React, { useEffect, useState } from "react";
+import { Text, View, Button } from "react-native";
+import { Question } from "../types/QuizQuestions";
+import { quizQuestions } from "../types/QuizQuestions";
 
-export default function StateMutipleChoiceQuizPage() {
-  return (
+export default function StateMultipleChoiceQuizApp() {
```
Comment (Kicchu02) [inline components/StateMultipleChoiceQuizPage.tsx]: Why was this name changed? Shouldn't it be `StateMultipleChoiceQuizPage`? Didn't we already establish the fact that the component name should be same as the file name? Why was this intentionally done?

**Review by Kicchu02 — CHANGES_REQUESTED** [review]
Nothing is changed in the PR. I have already [mentioned](https://github.com/Kicchu02/States/pull/10#pullrequestreview-2244384268) to address all the changes requested before asking for review.

Make sure you do not ask for review again without addressing all the changes.

File: components/StateMultipleChoiceQuizPage.tsx (hunk 6 above)
Comment (Kicchu02) [reply components/StateMultipleChoiceQuizPage.tsx]: If you are putting something in between, then the opening and closing tag is required. Example:
```tsx
<Text>Hello World</Text>
```
But if there is nothing in between, then it is convention to use a self closing tag. Example:
```tsx
<WelcomePage/>
```

File: components/StateMultipleChoiceQuizPage.tsx (hunk 10 above)
Comment (Kicchu02) [reply components/StateMultipleChoiceQuizPage.tsx]: The number of questions selected for the game should be lesser than the total number of questions, as per the design discussed offline.

**Review by Kicchu02 — APPROVED** [review]

**Pull request merged**